src/alr/alr-commands-init.adb

```diff
@@ -79,30 +81,84 @@ package body Alr.Commands.Init is
+         Put_Line ("   Style_Checks : Enabled_Kind := External (""" & Upper_Name & "_STYLE_CHECKS"", ""enabled"");");
+         Put_Line ("   Contracts_Checks : Enabled_Kind := External (""" & Upper_Name & "_CONTRACTS"", ""enabled"");");
+         Put_New_Line;
+         Put_Line ("   type Build_Kind is (""debug"", ""optimize"");");
+         Put_Line ("   Build_Mode : Build_Kind := External (""" & Upper_Name & "_BUILD_MODE"", ""debug"");");
+         Put_New_Line;
+         Put_Line ("   Compile_Checks_Switches := ();");
+         Put_Line ("   case Compile_Checks is");
+         Put_Line ("      when ""enabled"" =>");
+         Put_Line ("         Compile_Checks_Switches :=");
+         Put_Line ("           (""-gnatwa"",  -- All warnings");
+         Put_Line ("            ""-gnatVa"",  -- All validity checks");
+         Put_Line ("            ""-gnatwe""); -- Warnings as errors");
+         Put_Line ("      when others => null;");
+         Put_Line ("   end case;");
+         Put_New_Line;
+         Put_Line ("   Runtime_Checks_Switches := ();");
+         Put_Line ("   case Runtime_Checks is");
+         Put_Line ("      when ""enabled"" => null;");
+         Put_Line ("      when others =>");
+         Put_Line ("         Runtime_Checks_Switches :=");
+         Put_Line ("           (""-gnatp""); -- Supress checks");
+         Put_Line ("   end case;");
+         Put_New_Line;
+         Put_Line ("   Style_Checks_Switches := ();");
+         Put_Line ("   case Style_Checks is");
+         Put_Line ("      when ""enabled"" => null;");
+         Put_Line ("         Style_Checks_Switches :=");
+         Put_Line ("           (""-gnaty"",    -- Style checks");
+         Put_Line ("            ""-gnatyM80"", -- Maximum line length");
+         Put_Line ("            ""-gnatyO"");  -- Overriding subprograms explicitly marked as such");
+         Put_Line ("      when others => null;");
+         Put_Line ("   end case;");
+         Put_New_Line;
+         Put_Line ("   Contracts_Switches := ();");
+         Put_Line ("   case Contracts_Checks is");
+         Put_Line ("      when ""enabled"" => null;");
+         Put_Line ("         Contracts_Switches :=");
+         Put_Line ("           (""-gnata""); --  Enable assertions and contracts");
+         Put_Line ("      when others =>");
+         Put_Line ("   end case;");
+         Put_New_Line;
+         Put_Line ("   Build_Switches := ();");
+         Put_Line ("   case Build_Mode is");
+         Put_Line ("      when ""optimize"" =>");
+         Put_Line ("         Build_Switches := (""-O3"",     -- Optimization");
+         Put_Line ("                            ""-gnatn""); -- Enable inlining");
+         Put_Line ("      when ""debug"" =>");
+         Put_Line ("         Build_Switches := (""-g"",   -- Debug info");
+         Put_Line ("                            ""-O0""); -- No optimization");
```

Comment:
What do you think about using `-Og` instead of `-O0`? According to https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

> It is a better choice than -O0 for producing debuggable code because some compiler passes that collect debug information are disabled at -O0.

Personally, I haven't noticed anything ever but...

Reply:
Ok for me